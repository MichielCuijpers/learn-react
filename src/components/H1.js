import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';

function H1(props) {
  const { children: title } = props;

  return (
    <div>
      <h1 className="heading">{title}</h1>
    </div>
  );
}

// Specifies the default values for props:
H1.defaultProps = {
  children: 'Hello There',
};

H1.propTypes = {
  children: PropTypes.string,
};

export default hot(module)(H1);
