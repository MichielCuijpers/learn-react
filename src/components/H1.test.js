import React from 'react';
import { shallow } from 'enzyme';
import H1 from './H1';

// To use expect clause, see reference
// https://jestjs.io/docs/en/expect
// https://github.com/FormidableLabs/enzyme-matchers/tree/master/packages/jest-enzyme
describe('H1', () => {
  it('should have one with h1', () => {
    const component = shallow(<H1 />);
    expect(component.find('h1')).toHaveLength(1);
  });

  it('should have one with class `.title`', () => {
    const component = shallow(<H1 />);
    expect(component.find('.title')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const component = shallow(<H1 />);
    expect(component).toMatchSnapshot();
  });
});
