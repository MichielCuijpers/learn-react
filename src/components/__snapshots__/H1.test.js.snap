// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 should match snapshot 1`] = `
<div>
  <h1
    className="heading"
  >
    Hey There
  </h1>
</div>
`;
